src/core/meshcop/border_agent.cpp

```diff
@@ -0,0 +1,600 @@
+    responseHeader.SetPayloadMarker();
+    VerifyOrExit((message = mNetif.GetSecureCoapServer().NewMeshCoPMessage(responseHeader)) != NULL,
+                 error = kThreadError_NoBufs);    
+    
+    SuccessOrExit(error = CopyPayload(aMessage, message));
+
+    messageInfo.SetPeerAddr(mCommissionerAddr);
+    messageInfo.SetPeerPort(mCommissionerUdpPort);
+
+    SuccessOrExit(error = mNetif.GetSecureCoapServer().SendMessage(*message, messageInfo));
+
+    otLogInfoMeshCoP(GetInstance(), "sent MGMT_COMMISSIONER_SET response to commissioner");
+
+exit:
+    (void) aMessageInfo;
+    if (error != kThreadError_None && message != NULL)
+    {
+        message->Free();
+    }
+    otLogFuncExit();
+}
+
+void BorderAgent::HandleRelayReceive(void *aContext, otCoapHeader *aHeader, otMessage *aMessage,
+                                     const otMessageInfo *aMessageInfo)
+{
+    static_cast<BorderAgent *>(aContext)->HandleRelayReceive(
+        *static_cast<Coap::Header *>(aHeader), *static_cast<Message *>(aMessage),
+        *static_cast<const Ip6::MessageInfo *>(aMessageInfo));
+}
+
+void BorderAgent::HandleRelayReceive(Coap::Header &aHeader, Message &aMessage,
+                                     const Ip6::MessageInfo &aMessageInfo)
+{
+    ThreadError error = kThreadError_None;
+    Coap::Header header;
+    Message *message = NULL;
+    Ip6::MessageInfo messageInfo;
+
+    otLogFuncEntry();
+    otLogInfoMeshCoP(GetInstance(), "received Relay Recevie from Joiner Router");
+
+    header.Init(kCoapTypeConfirmable, kCoapRequestPost);
+    header.SetToken(aHeader.GetToken(), aHeader.GetTokenLength());
+    header.AppendUriPathOptions(OPENTHREAD_URI_RELAY_RX);
+    header.SetPayloadMarker();
+    
+    VerifyOrExit((message = mNetif.GetSecureCoapServer().NewMeshCoPMessage(header)) != NULL, error = kThreadError_NoBufs);
+
+    SuccessOrExit(error = CopyPayload(aMessage, message));
+
+    messageInfo.SetPeerAddr(mCommissionerAddr);
+
+    otLogInfoMeshCoP(GetInstance(), "send Relay Receive to Commissioner");
+
+    SendCommissionerRelayReceive(*message, messageInfo);
+
+    
+exit:
+    (void)aMessageInfo;
+    if (error != kThreadError_None && message != NULL)
+    {
+        message->Free();
+    }
+    otLogFuncExit();
+}
+
+
+ThreadError BorderAgent::SendCommissionerRelayReceive(Message &aMessage, const Ip6::MessageInfo &aMessageInfo)
+{
+    ThreadError error = kThreadError_None;
+
+    otLogFuncEntry();
+
+    SuccessOrExit(error = mNetif.GetSecureCoapServer().SendMessage(aMessage, aMessageInfo));
+
+    otLogInfoMeshCoP(GetInstance(), "sent Realy Receive to Commissioner");
+
+exit:
+    (void) aMessageInfo;
+    otLogFuncExitErr(error);
+    return error;
+}
+
+void BorderAgent::HandleTimer(void *aContext)
+{
+    static_cast<BorderAgent *>(aContext)->HandleTimer();
+}
+
+void BorderAgent::HandleTimer(void)
+{
+    ThreadError error = kThreadError_Error;
+    PSKcTlv *pskc;
+    char aThreadPSKc[] = 
```

Comment:
What you've said is reasonable here but, i want a variable to record the pskc default value, if the border agent cannot obtain the pskc from the Leader (data response tlv), it can use the default value to start DTLS server.